onedal/datatypes/tests/test_data.py

```diff
@@ -167,69 +230,150 @@ def test_conversion_to_table(dtype):
+    X_table, result_responses_table, result_responses_df = alg.fit(X)
+
+    assert hasattr(X_table, "__sycl_usm_array_interface__")
+    assert hasattr(result_responses_table, "__sycl_usm_array_interface__")
+    assert hasattr(result_responses_df, "__sycl_usm_array_interface__")
+    assert hasattr(X, "__sycl_usm_array_interface__")
+    _assert_sua_iface_fields(X, X_table)
+
+    # Work around for saving compute-follows-data execution
+    # for CPU sycl context requires cause additional memory
+    # allocation using the same queue.
+    skip_data_0 = True if queue.sycl_device.is_cpu else False
+    # Onedal return table's syclobj is empty for CPU inputs.
+    skip_syclobj = True if queue.sycl_device.is_cpu else False
+    # TODO:
+    # investigate why __sycl_usm_array_interface__["data"][1] is changed
+    # after conversion from onedal table to sua array.
+    # Test is not turned off because of this. Only check is skipped.
+    skip_data_1 = True
+    _assert_sua_iface_fields(
+        result_responses_df,
+        result_responses_table,
+        skip_data_0=skip_data_0,
+        skip_data_1=skip_data_1,
+        skip_syclobj=skip_syclobj,
+    )
+    assert X.sycl_queue == result_responses_df.sycl_queue
+    if order == "F":
+        assert X.flags.f_contiguous == result_responses_df.flags.f_contiguous
+    else:
+        assert X.flags.c_contiguous == result_responses_df.flags.c_contiguous
+    # 1D output expected to have the same c_contiguous and f_contiguous flag values.
     assert (
-        x_dpt.__sycl_usm_array_interface__["data"][0]
-        == x_dpt_from_table.__sycl_usm_array_interface__["data"][0]
+        result_responses_df.flags.c_contiguous == result_responses_df.flags.f_contiguous
     )
-    assert x_dpt.shape == x_dpt_from_table.shape
-    assert x_dpt.strides == x_dpt_from_table.strides
-    assert x_dpt.dtype == x_dpt_from_table.dtype
-    assert x_dpt.flags.f_contiguous
-    assert x_dpt_from_table.flags.f_contiguous
+
+
+@pytest.mark.skipif(
+    not _is_dpc_backend,
+    reason="__sycl_usm_array_interface__ support requires DPC backend.",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue", get_dataframes_and_queues("dpctl, dpnp", "cpu, gpu")
+)
+@pytest.mark.parametrize("data_shape", unsupported_data_shapes)
+def test_sua_iface_interop_invalid_shape(dataframe, queue, data_shape):
+    X = np.zeros(data_shape)
+    X = _convert_to_dataframe(X, sycl_queue=queue, target_df=dataframe)
+    sua_iface, _, _ = _get_sycl_namespace(X)
+
+    expected_err_msg = (
+        "Unable to convert from SUA interface: only 1D & 2D tensors are allowed"
+    )
+    with pytest.raises(ValueError, match=expected_err_msg):
+        to_table(X, sua_iface=sua_iface)
+
+
+@pytest.mark.skipif(
+    not _is_dpc_backend,
+    reason="__sycl_usm_array_interface__ support requires DPC backend.",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue", get_dataframes_and_queues("dpctl, dpnp", "cpu, gpu")
```

Comment:
different string strategy than before with the comma and space in ```"dpctl, dpnp"``` throughout the PR. Is this purposeful?

Reply:
not in purpose. will change.